services/121-service/src/payments/payments.service.ts

```diff
@@ -219,43 +219,27 @@ export class PaymentsService {
   private async getRegistrationsForPayment(
     programId: number,
     payment: number,
-    referenceId?: string,
+    referenceIds?: ReferenceIdsDto,
   ): Promise<RegistrationEntity[]> {
-    const knownPayment = await this.transactionRepository.findOne({
-      where: { payment: payment },
-    });
-    let failedRegistrations;
-    if (knownPayment) {
-      const failedReferenceIds = (
-        await this.getFailedTransactions(programId, payment)
-      ).map(t => t.referenceId);
-      failedRegistrations = await this.registrationRepository.find({
-        where: { referenceId: In(failedReferenceIds) },
+    if (referenceIds) {
+      return await this.registrationRepository.find({
+        where: { referenceId: In(JSON.parse(referenceIds['referenceIds'])) },
```

Comment:
Why is this `JSON.parse` necessary? If we never `stringify` it, we won't need to `parse` it, right?